Frankie DSM fwrd Prog (more improvements)

frankiearzu opened the pull request:
[#751](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/issues/751)

This will be my lastpull  request for some time.. I think i got the tool with the functionality i want.
Tested with AR631 and Heli  FC6250hx.  Updated Rx Simulation with ALL menus.

1. Fix problems when text contradictions between Menu/Line Headers and List Values (Ex.. Same TextId showing different text on Spektrum radios).. Solution: keep List_Text values separate
3. Show Images of RX orientations (Initial Setup)
4. Able to Hack getting into Initial Setup who was failing with original code on AR631/AR637
5. Custumize the way Flight Mode reports the Value on Screen  (Instead of Flight Mode 0, show Flight Mode 1".. In Helicopter RX Flight Mode 2 is Stunt-1, etc.

pfeerick commented:
I just wanted to thank you for all the work you have done on this... I've just pulled my AR637T out of the drawer, rebound it, and first used `main` and then tried this PRs code to set it up again. This PR fixed being able to run through the initial setup, as some of the strings were missing. When I first got this RX, I needed to use a flying buddies DX7 to get through the first time setup, and then the forward programming Lua was mostly functional. This is not only prettier, and more functional, but upon doing a factory reset I was also able to go through the first-time setup via the MPM without a hitch. 

Other than the occasional "Unable to read menu lines from RX" (perhaps as RX is too close to TX?), and what seemed to be a bit of stickyness with setting the FM switch (it kept resetting to AUX1 after I changed it even though I had moved and set it to AUX2, until I hit DEF and then moved the switch again) this seems to be working perfectly - and I'm more inclinded to think those issues are possibly more to do with the telem link, RX firmware and user error than this code. ;)

frankiearzu commented:
Thank you Peter. I notice the Aux1 stickyness.  After a value is changed, the RX validates it.. if is invalid, it internally correct it to a valid value.Even that is a 6ch RX, internaly can receive more.. will check that is the RX.  The /LOGS/dsm_log.txt will show if the value is been corrected by the RX. you will see that we send an update of the value, and later we ask the RX to validate the value. the RX will return an update overriding your value if invalid.
As for "Unable to read menu lines...", some menus returned an unknown type of line that has not been completly understood what is it for.. Some menus gets stuck in a infinite look retriving this menu lines.  When this happens, and the program exits, can you check the /LOGS/dsm_log.txt  and go to the end.. if you see "Unknown_0x05" lines.. we got into that loop.   If the same menu always fails in this way, send me the log, this could be a menu that needs a hack over how to retrive it (a bit different values than the normal flow).You can see in the log if we are aplying the hack. is written there.
Thank you for testing,Frankie   
    On Saturday, November 19, 2022 at 09:25:08 PM CST, Peter Feerick ***@***.***> wrote:  
 
 


I just wanted to thank you for all the work you have done on this... I've just pulled my AR637T out of the drawer, rebound it, and first used main and then tried this PRs code to set it up again. This PR fixed being able to run through the initial setup, as some of the strings were missing. When I first got this RX, I needed to use a flying buddies DX7 to get through the first time setup, and then the forward programming Lua was mostly functional. This is not only prettier, and more functional, but upon doing a factory reset I was also able to go through the first-time setup via the MPM without a hitch.

Other than the occasional "Unable to read menu lines from RX" (perhaps as RX is too close to TX?), and what seemed to be a bit of stickyness with setting the FM switch (it kept resetting to AUX1 after I changed it even though I had moved and set it to AUX2, until I hit DEF and then moved the switch again) this seems to be working perfectly - and I'm more inclinded to think those issues are possibly more to do with the telem link, RX firmware and user error than this code. ;)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

frankiearzu commented:
Peter,I found how to use the Aux2-Aux7 for FM Channel.   When you are edditing the channel field.. you have to also toggle the switch that you use to change flight mode (and configured for channel 7-9).. In that way, the RX is aware that the channel is in use, and will not reject the value. Actually while editting, if you have any other switch in a lower channel and you use it.. it will select the value inside the field by activating the switch.
Frankie


    On Sunday, November 20, 2022 at 12:48:24 PM CST, Francisco Arzu ***@***.***> wrote:  
 
  Thank you Peter. I notice the Aux1 stickyness.  After a value is changed, the RX validates it.. if is invalid, it internally correct it to a valid value.Even that is a 6ch RX, internaly can receive more.. will check that is the RX.  The /LOGS/dsm_log.txt will show if the value is been corrected by the RX. you will see that we send an update of the value, and later we ask the RX to validate the value. the RX will return an update overriding your value if invalid.
As for "Unable to read menu lines...", some menus returned an unknown type of line that has not been completly understood what is it for.. Some menus gets stuck in a infinite look retriving this menu lines.  When this happens, and the program exits, can you check the /LOGS/dsm_log.txt  and go to the end.. if you see "Unknown_0x05" lines.. we got into that loop.   If the same menu always fails in this way, send me the log, this could be a menu that needs a hack over how to retrive it (a bit different values than the normal flow).You can see in the log if we are aplying the hack. is written there.
Thank you for testing,Frankie   
    On Saturday, November 19, 2022 at 09:25:08 PM CST, Peter Feerick ***@***.***> wrote:  
 
 


I just wanted to thank you for all the work you have done on this... I've just pulled my AR637T out of the drawer, rebound it, and first used main and then tried this PRs code to set it up again. This PR fixed being able to run through the initial setup, as some of the strings were missing. When I first got this RX, I needed to use a flying buddies DX7 to get through the first time setup, and then the forward programming Lua was mostly functional. This is not only prettier, and more functional, but upon doing a factory reset I was also able to go through the first-time setup via the MPM without a hitch.

Other than the occasional "Unable to read menu lines from RX" (perhaps as RX is too close to TX?), and what seemed to be a bit of stickyness with setting the FM switch (it kept resetting to AUX1 after I changed it even though I had moved and set it to AUX2, until I hit DEF and then moved the switch again) this seems to be working perfectly - and I'm more inclinded to think those issues are possibly more to do with the telem link, RX firmware and user error than this code. ;)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
